pycbc/types/timeseries.py

```diff
@@ -795,6 +795,22 @@ def fir_zero_filter(self, coeff):
         from pycbc.filter import fir_zero_filter
         return self._return(fir_zero_filter(coeff, self))
 
+    def resample(self, delta_t):
+        """ Resample this time series to the new delta_t
+        
+        Parameters
+        -----------
+        delta_t: float
+            The time step to resample the times series to.
+            
+        Returns
+        -------
+        resampled_ts: pycbc.types.TimeSeries
+            The resample timeseries at the new time interval delta_t.
+        """
+        from pycbc.filter import resample_to_delta_t
```

Comment:
Do we want imports within the method itself and not at the top?

Reply:
We need to delay the import to avoid a cyclic import here.

Reply:
Okay, all good. Thanks!